File: category-backend/src/main/java/com/namit/categorybackend/product/service/ProductService.java
package com.namit.categorybackend.product.service;


import com.namit.categorybackend.common.response.PagedResponse;
import com.namit.categorybackend.product.dto.ProductRequest;
import com.namit.categorybackend.product.dto.ProductResponse;
import com.namit.categorybackend.product.entity.Product;

public interface ProductService {

    // Creates a new product
    ProductResponse createProduct(ProductRequest request);

    // Get all products
    PagedResponse<ProductResponse> getAllProducts(int page , int size , String status);

    // Get product by ID
    ProductResponse getProductById( Long id);

    // Toggle product staus (Active <-> Inactive)
    ProductResponse toggleProductStatus(Long id);

    // Update Product
    ProductResponse updateProduct(Long id , ProductRequest request);
}

File: category-frontend/src/services/productService.js
import api from "../api/axios";

export const getProducts = (page = 0, size = 10, status = "active") =>
    api.get(`/products?page=${page}&size=${size}&status=${status}`);

export const getPublicProducts = (page = 0, size = 12) =>
    api.get(`/products/public?page=${page}&size=${size}`);

export const getProductById = (id) => api.get(`/products/${id}`);

export const createProduct = (data) => api.post("/products", data);

export const updateProduct = (id, data) => api.put(`/products/${id}`, data);

export const toggleProductStatus = (id) => api.patch(`/products/${id}/toggle`);
